src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -110,14 +114,40 @@ DlgPrefMixer::DlgPrefMixer(
     connect(SliderXFader,
             QOverload<int>::of(&QSlider::valueChanged),
             this,
-            &DlgPrefMixer::slotUpdateXFader);
-    connect(SliderXFader, &QSlider::sliderMoved, this, &DlgPrefMixer::slotUpdateXFader);
-    connect(SliderXFader, &QSlider::sliderReleased, this, &DlgPrefMixer::slotUpdateXFader);
-    connect(radioButtonAdditive, &QRadioButton::clicked, this, &DlgPrefMixer::slotUpdateXFader);
+            &DlgPrefMixer::slotXFaderWidgetsChanged);
+    connect(SliderXFader, &QSlider::sliderMoved, this, &DlgPrefMixer::slotXFaderWidgetsChanged);
+    connect(SliderXFader, &QSlider::sliderReleased, this, &DlgPrefMixer::slotXFaderWidgetsChanged);
+    connect(radioButtonAdditive,
+            &QRadioButton::clicked,
+            this,
+            &DlgPrefMixer::slotXFaderWidgetsChanged);
     connect(radioButtonConstantPower,
             &QRadioButton::clicked,
             this,
-            &DlgPrefMixer::slotUpdateXFader);
+            &DlgPrefMixer::slotXFaderWidgetsChanged);
+    connect(checkBoxReverse,
+            // TODO deprecated after Qt 6.9, replace with checkStateChanged(Qt::CheckState)
+            &QCheckBox::stateChanged,
+            this,
+            &DlgPrefMixer::slotXFaderReverseBoxToggled);
+
+    m_xfModeCO->connectValueChanged(this, &DlgPrefMixer::slotXFaderControlChanged);
+    m_xfCurveCO->connectValueChanged(this, &DlgPrefMixer::slotXFaderControlChanged);
+    m_xfReverseCO->connectValueChanged(this, &DlgPrefMixer::slotXFaderControlChanged);
+    // When e.g. controller mappings rapidly set multiple xfader controls we may
+    // get concurrent calls of slotUpdateXFaderFromConfigOrControls().
+    // In slotXFaderControlChanged() we start a timer to avoid that, only the last
+    // calls th update method.
+    connect(&m_xfaderControlUpdateTimer,
+            &QTimer::timeout,
+            this,
+            [this]() {
+                slotUpdateXFaderFromConfigOrControls();
+                // store new settings in config
+                applyXFader();
+            });
+    m_xfaderControlUpdateTimer.setSingleShot(true);
+    m_xfaderControlUpdateTimer.setInterval(200);
```

Comment:
I doubt this is a super good solution. Do we actually get truly concurrent calls (in different threads) or just rapid calls in quick succession? Which part is actually so slow that it warrants the debouncing?

Reply:
If you know what attributes exactly influence the graphic, you can compare the old and new state and reject calls that result in the same or similar values. That seems like a reasonable way to do this without having to introduce a hacky timer.

Reply:
Yeah, "concurrent" is the wrong term, it's indeed rapid calls in quick succession.
I saw that I switched the 3-state xfader toggle on the S4mk3 (0 -> 1) and it caused some update misses / reverted the mode state.
But I think is might have been happening in an earlier stage only.

I'll need to rework this further anyway, like more separation of updates from control changes / widget changes. (maybe some blockSignals() will already fix it)